### PR TITLE
update(rules): add containerd_activities check to Clear Log Activities rule

### PR DESCRIPTION
**Pull request by brennoo**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area rules

> /area registry

> /area build

> /area documentation

**Proposed rule [maturity level](https://github.com/falcosecurity/rules/blob/main/CONTRIBUTING.md#maturity-levels)**

> Uncomment one (or more) `/area <>` lines (only for PRs that add or modify rules):

/area maturity-stable

> /area maturity-incubating

> /area maturity-sandbox

> /area maturity-deprecated

**What this PR does / why we need it**:

This PR adds the `containerd_activities` check to the `Clear Log Activities` rule as we have noticed that the rule is being triggered by `containerd` activities in the host, example json output:
```json
{"uuid":"72b27712-9a80-4180-ab2f-c7d298114fb9","output":"18:31:35.698169341: Warning Log files were tampered (file=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/338/fs/var/log/dpkg.log evt_type=openat user=root user_uid=0 user_loginuid=-1 process=containerd proc_exepath=/usr/bin/containerd parent=systemd command=containerd terminal=0 exe_flags=O_TRUNC|O_CREAT|O_WRONLY|O_CLOEXEC|O_F_CREATED container_id=host container_image=<NA> container_image_tag=<NA> container_name=host k8s_ns=<NA> k8s_pod_name=<NA>)","priority":"Warning","rule":"Clear Log Activities","time":"2023-09-27T18:31:35.698169341Z","source":"syscall","output_fields":{"container.id":"host","container.image.repository":null,"container.image.tag":null,"container.name":"host","evt.arg.flags":"O_TRUNC|O_CREAT|O_WRONLY|O_CLOEXEC|O_F_CREATED","evt.time":1695839495698169300,"evt.type":"openat","fd.name":"/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/338/fs/var/log/dpkg.log","k8s.ns.name":null,"k8s.pod.name":null,"proc.cmdline":"containerd","proc.exepath":"/usr/bin/containerd","proc.name":"containerd","proc.pname":"systemd","proc.tty":0,"user.loginuid":-1,"user.name":"root","user.uid":0},"hostname":"production-us-cluster-0-falco-sf9n5","tags":["NIST_800-53_AU-10","T1070","container","filesystem","host","maturity_stable","mitre_defense_evasion"]}
```

**Which issue(s) this PR fixes**: I haven't raised one as it seems quite simple

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by incertum — COMMENTED** [review]
@darryk10 for a second opinion. I would propose to keep this rule for host and container workloads and instead add additional tuning to it?

**Comment by darryk10** [comment]
could we perhaps exclude the specific fd path.
Excluding something like (proc.name=containerd and fd.name endswith /var/log/dpkg.log)
As far as I know the noise is just on this file. It might not worth it reduce the overall rule use case for one file.

**Comment by brennoo** [comment]
hi there, thanks for reviewing my first PR :-)

> could we perhaps exclude the specific fd path. Excluding something like (proc.name=containerd and fd.name endswith /var/log/dpkg.log) As far as I know the noise is just on this file. It might not worth it reduce the overall rule use case for one file.

this is what I did in my local rules `(fd.path startswith "/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/")` but it seems `containerd` also touch other paths, so far only one extra prefix:
```bash
$ jq '.[].output_fields | ."fd.name"' falco_events.json | sort | uniq -c | sort -rn | head
   1 "/var/lib/containerd/tmpmounts/containerd-mount991927798/var/log/yum.log"
   1 "/var/lib/containerd/tmpmounts/containerd-mount980514857/var/log/dpkg.log"
   1 "/var/lib/containerd/tmpmounts/containerd-mount83416718/var/log/yum.log"
   1 "/var/lib/containerd/tmpmounts/containerd-mount826255972/var/log/dpkg.log"
   1 "/var/lib/containerd/tmpmounts/containerd-mount813808526/var/log/anaconda/syslog"
   1 "/var/lib/containerd/tmpmounts/containerd-mount807056178/var/log/yum.log"
   1 "/var/lib/containerd/tmpmounts/containerd-mount709329761/var/log/yum.log"
   1 "/var/lib/containerd/tmpmounts/containerd-mount661173829/var/log/dpkg.log"
   1 "/var/lib/containerd/tmpmounts/containerd-mount61672547/var/log/dpkg.log"
   1 "/var/lib/containerd/tmpmounts/containerd-mount587022222/var/log/dpkg.log"
```
I'm adding this prefix to `allowed_clear_log_files` and will monitor during the day, for some reason I had the impression that this rule was meant for containers only but it makes sense it would also good to use it for host detection.

**Comment by brennoo** [comment]
OK, this macro fixed the false positives:
```yaml
- macro: allowed_clear_log_files
  condition: (proc.name=containerd and (fd.name startswith "/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/" or
              fd.name startswith "/var/lib/containerd/tmpmounts/"))
```
is this something you would consider adding upstream as a dedicated macro or such cases must be handled locally?

**Comment by incertum** [comment]
Yes from my side +1, ty!

**Comment by darryk10** [comment]
+1 from my side as well! Thanks @brennoo

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -921,6 +921,10 @@
               container.image.repository endswith "openshift3/ose-logging-fluentd" or
               container.image.repository endswith "containernetworking/azure-npm")
 
+- macro: containerd_activities
+  condition: proc.name=containerd and (fd.name startswith "/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/" or
```
Comment (incertum) [inline rules/falco_rules.yaml:925]: Just a super minor change request:

Add parentheses to be future proof if we were to keep adding on e.g. `and not (containerd_activities or other_macro)`

```
(proc.name=containerd and (fd.name startswith "/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/" or
                                       fd.name startswith "/var/lib/containerd/tmpmounts/"))
```

**Comment by incertum** [comment]
@jasondellaluce what is current guidance re failing CI because of the required engine version format cutover?Thanks!

**Review by incertum — APPROVED** [review]
/approve

**Pull request merged**